File: backend/app/schemas/qa.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class QAIssue(BaseModel):
    type: Literal[
        "missing_evidence",
        "weak_evidence",
        "unsupported_claim",
        "contradiction",
        "outdated_source",
        "schema_incomplete",
        "logic_gap",
        "writing_issue",
    ]
    severity: Literal["low", "medium", "high"]
    message: str
    related_claim_id: int | None = None
    related_competitor: str | None = None
    related_dimension: str | None = None
    suggested_action: Literal["recollect", "reanalyze", "rewrite", "ignore"]
    target_node: str | None = None
    search_query: str | None = None


class QAResultPayload(BaseModel):
    passed: bool
    score: float = Field(ge=0, le=1)
    issues: list[QAIssue] = []
    next_action: Literal["end", "recollect", "reanalyze", "rewrite"] = "end"
    target_nodes: list[str] = []
    revision_reason: str | None = None
    followup_queries: list[str] = []
    revision_round: int = 0


class QAResultItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    report_id: int | None
    passed: bool
    score: Decimal | None
    issues: list[dict]
    next_action: str = "end"
    next_action_label: str = "无需返工"
    target_nodes: list[str] = []
    target_node_labels: list[str] = []
    revision_reason: str | None = None
    revision_round: int = 0
    created_at: datetime


class QAResultResponse(BaseModel):
    qa_result: QAResultItem | None

File: frontend/src/types/qa.ts
export interface QAIssue {
  type: string
  severity: string
  message: string
  related_claim_id?: number | null
  related_competitor?: string | null
  related_dimension?: string | null
  suggested_action?: string
  suggested_action_label?: string
  target_node?: string | null
  target_node_label?: string | null
  search_query?: string | null
  type_label?: string
  severity_label?: string
}

export interface DimensionQAScore {
  target_node: string
  dimension_key?: string | null
  dimension_label?: string | null
  score?: number | string | null
  passed?: boolean
  issues?: QAIssue[]
  qa_scope?: string
  qa_scope_label?: string
  revision_round?: number
}

export interface QAResult {
  id: number
  task_id: number
  report_id?: number | null
  passed: boolean
  score?: string | number | null
  issues: QAIssue[]
  dimension_scores?: DimensionQAScore[]
  current_dimension_scores?: DimensionQAScore[]
  next_action?: string
  next_action_label?: string
  target_nodes?: string[]
  target_node_labels?: string[]
  qa_scope?: string
  qa_scope_label?: string
  revision_reason?: string | null
  revision_round?: number
  created_at: string
}
